include/ur_client_library/comm/tcp_socket.h

```diff
@@ -50,7 +50,7 @@ class TCPSocket
 private:
   std::atomic<int> socket_fd_;
   std::atomic<SocketState> state_;
-  std::chrono::seconds reconnection_time_;
+  std::chrono::milliseconds reconnection_time_;
```

Comment:
I know that reconnection_time was not added in this PR, but I suggest that we add the reconnection_time as an input argument to the setup function instead of a member variable. Then max_num_tries and reconnection_time can be modified in the same way and both variables are only used within the setup function.

Reply:
That sounds reasonable.

Reply:
I've added it to the `setup` method and added a deprecation notice. Due to being implemented as a default parameter value I couldn't properly overload in order to keep the old and current API stable. Hence, I added a manual conflict resolution to choose the value set through the deprecated API with an additional deprecation notice.